driver/vm.go

```diff
@@ -337,8 +339,22 @@ func (vm *VirtualMachine) PowerOn() error {
 	return err
 }
 
-func (vm *VirtualMachine) WaitForIP(ctx context.Context) (string, error) {
-	return vm.vm.WaitForIP(ctx)
+func (vm *VirtualMachine) WaitForIP(ctx context.Context) (ip string, err error) {
+	collector := property.DefaultCollector(vm.vm.Client())
+	err = property.Wait(ctx, collector, vm.vm.Reference(), []string{"guest.ipAddress"}, func(pc []types.PropertyChange) bool {
+		for _, c := range pc {
+			if c.Name == "guest.ipAddress" && c.Op == types.PropertyChangeOpAssign && c.Val != nil {
+				ipCandidate := net.ParseIP(c.Val.(string))
+				if ipCandidate != nil && !ipCandidate.IsLinkLocalUnicast() {
```

Comment:
Ensure the value received is both a valid IP, as well as not [link-local unicast](https://en.wikipedia.org/wiki/Link-local_address#IPv6).